language/functional_tests/tests/testsuite/borrow_tests/mutate_with_borrowed_loc_resource_invalid.mvir

```diff
@@ -0,0 +1,17 @@
+module M {
+    resource X {}
+    t1() {
+        let x: Self.X;
+        let y: &Self.X;
+        x = X {};
+        y = &x;
+        x = X {};
+        _ = move(y);
+        X {} = move(x);
+        return;
+    }
+
+}
+
+// check: VerificationFailure
+// check: STLOC_UNSAFE_TO_DESTROY_ERROR
```

Comment:
One more test doing something like:

```
x = X { f: 2 }
y = &x;
z = &x.f
y = X { f: 7 }
```

might be useful?

Reply:
I added this test as it is helpful for testing borrows being transitioned upwards on field borrows. 

But there is no other shenanigans going on here (as you might have indicated with your `Vector` comment)